tests/test_events/test_api.py

```diff
@@ -198,16 +198,36 @@ def test_singleton_pattern_working_correctly_for_event_api(self, mock_send: Magi
             ]
         )
 
-    @patch.object(Boto3Client, "eventbridge", MagicMock())
-    def test_second_send_clears_everything(self) -> None:
+    @patch.object(Boto3Client, "eventbridge")
+    def test_second_send_does_not_clear_everything(self, mock_send: MagicMock) -> None:
+        mock_send.put_events.side_effect = NotADirectoryError
         event = MyTestEvent({"x": 1})
         self.event_api.register(event)
+        with pytest.raises(RuntimeError):
+            self.event_api.send()
+        mock_send.put_events.side_effect = None
```

Comment:
Also, you can declare the side effect at the beginning:
```suggestion
        mock_send.put_events.side_effect = [NotADirectoryError, None]
        event = MyTestEvent({"x": 1})

        self.event_api.register(event)
        with pytest.raises(RuntimeError):
            self.event_api.send()
```